### ISSUE TITLE
Custom type not Working with TYPO3 10.4 

### ISSUE THREAD

**Issue by Megafry** [georgringer/news#1306]
## Bug Report

**Current Behavior**
I have an extension which adds a new type, this works fine with TYPO3 9.5 but unfortunately not with TYPO3 10.4.

**Expected behavior/output**
If the record type is ex: 3 it should use my own model (as explained in the [doc](https://docs.typo3.org/p/georgringer/news/master/en-us/DeveloperManual/ExtendNews/AddCustomType/Index.html)). 
I have created a small extension, everything works fine with TYPO3 9.5, but not with TYPO3 10.4.
[My Extension](https://github.com/Megafry/mxn_team) 

My stackoverflow ticket: [#62510127](https://stackoverflow.com/questions/62510127/typo3-v10-extnews-custom-type-and-persistence-mapping)

**Environment**
- TYPO3 version(s): [10.4.4]
- news version: [8.3.0]
- Is your TYPO3 installation set up with Composer (Composer Mode): [no]

**Possible Solution**
i migrated my persistence settings as described in this example [Link](https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/10.0/Breaking-87623-ReplaceConfigpersistenceclassesTyposcriptConfiguration.html) 
I have the impression something is still missing, but i don't know what.

Thanks for any help

**Comment by DavidBruchmann** [comment]
Mappin is done different in Version 10.
This is only for version 9 or earlier:
```
config.tx_extbase.persistence.classes {
    GeorgRinger\News\Domain\Model\News {
        subclasses {
            3 = Vendor\ExtName\Domain\Model\MyCustomNewsType
        }
    }

    Vendor\ExtName\Domain\Model\MyCustomNewsType {
        mapping {
            recordType = 3
            tableName = tx_news_domain_model_news
        }
    }
}
```
The new way how to do it is described here:
https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/10.0/Breaking-87623-ReplaceConfigpersistenceclassesTyposcriptConfiguration.html

**Comment by Megafry** [comment]
Hi
Yes, this is what i did:
https://github.com/cyberhouse/t3ext-newsauthor/blob/master/ext_localconf.php

but I'm not sure if that's done right, any idea?

**Comment by DavidBruchmann** [comment]
You've to create the file `EXT:your_extension/Configuration/Extbase/Persistence/Classes.php` and add some mapping probably.
For TYPO3 version 9 it was done in the file `EXT:your_extension/ext_conf_template.txt`.
The syntax is different but it's explained if you look at the linked page again.

**Comment by Megafry** [comment]
Hi David
Yes, this is what i did with my test Extension, but the config is ignored.
 
[My Test-Extension ](https://github.com/Megafry/mxn_team/blob/master/Configuration/Extbase/Persistence/Classes.php)

**Comment by Typolist** [comment]
Hi,

test this configuration, it might work:

mxn_team/Configuration/TCA/Overrides/tx_news_domain_model_news.php:

```
<?php
defined('TYPO3_MODE') or die();

// Create locallang file (...)
**$ll = 'LLL:EXT:mxn_team/Resources/Private/Language/locallang_db.xlf:tx_mxn_team_domain_model_news.';**

$fields = [
    'card' => [
        'exclude' => true,
        'label' => 'LLL:EXT:mxn_team/Resources/Private/Language/locallang_db.xlf:tx_mxnteam_domain_model_team.card',
        'config' => [
            'type' => 'input',
            'size' => 30,
            'eval' => 'trim'
        ],
    ],
];

// 1.
\TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addTCAcolumns('tx_news_domain_model_news', $fields);
// 2.
\TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addToAllTCAtypes('tx_news_domain_model_news', 'card');
// 3. (**set here from mxn_team/ext_tables.php**)
\TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addStaticFile('mxn_team', 'Configuration/TypoScript', 'mxn_team');

// Set title and icon (example below) for the news type 
$GLOBALS['TCA']['tx_news_domain_model_news']['columns']['type']['config']['items']['3'] =
  [**$ll . 'card_title'**, 3,  '**ext-news-type-card**'] ;

// Set type configuration (**copied from last news extension**) and gradually delete individual areas that should not be displayed in the backend form
$GLOBALS['TCA']['tx_news_domain_model_news']['types']['3'] =
		[	
			'showitem' => '
					--palette--;;paletteCore,title,--palette--;;paletteSlug,teaser,
					--palette--;;paletteDate,
					bodytext,
				--div--;' . $llNews . 'tx_news_domain_model_news.content_elements,
					content_elements,
				--div--;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:tabs.media,
					fal_media,fal_related_files,
				--div--;LLL:EXT:core/Resources/Private/Language/Form/locallang_tabs.xlf:categories,
					categories,
				--div--;' . $llNews . 'tx_news_domain_model_news.tabs.relations,
					related,related_from,
					related_links,tags,
				--div--;LLL:EXT:frontend/Resources/Private/Language/locallang_tca.xlf:pages.tabs.metadata,
					--palette--;LLL:EXT:frontend/Resources/Private/Language/locallang_tca.xlf:pages.palettes.editorial;paletteAuthor,
					--palette--;LLL:EXT:frontend/Resources/Private/Language/locallang_tca.xlf:pages.palettes.metatags;metatags,
					--palette--;' . $llNews . 'tx_news_domain_model_news.palettes.alternativeTitles;alternativeTitles,
				--div--;LLL:EXT:core/Resources/Private/Language/Form/locallang_tabs.xlf:language,
					--palette--;;paletteLanguage,
				--div--;LLL:EXT:core/Resources/Private/Language/Form/locallang_tabs.xlf:access,
					--palette--;;paletteHidden,
					--palette--;;paletteAccess,
				--div--;' . $llNews . 'notes,
					notes,
				--div--;LLL:EXT:frontend/Resources/Private/Language/locallang_tca.xlf:pages.tabs.extended,**card**'
		];
```

// Example to register the icon (in ext_localconf.php)

```
// ...

if (TYPO3_MODE === 'BE') {	 
			/** @var \TYPO3\CMS\Core\Imaging\IconRegistry $iconRegistry */
			$iconRegistry = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance(\TYPO3\CMS\Core\Imaging\IconRegistry::class);
			
			$icons = [
				'ext-news-type-card' => 'mxn_team_domain_model_news_type3.svg'
				// ... more icons here ... comma separated
			];
			
			foreach ($icons as $identifier => $path) {
				$iconRegistry->registerIcon(
					$identifier,
					\TYPO3\CMS\Core\Imaging\IconProvider\SvgIconProvider::class,
					['source' => 'EXT:mxn_team/Resources/Public/Icons/Types/' . $path]
				);
			}
		}
```
I hope this helps!

With best regards
Typolist

**Comment by DavidBruchmann** [comment]
@Megafry entering an integer as `recordType` is quite strange for me, usually there has to be a model-class.  
Even if you have only one value to return this has to be returned probably by a class and can't assigned directly.

**Comment by Megafry** [comment]
Hi Typolist
Thanks for your help, but my problem is not to register the Icons, my problem is to make tx_news us my model if the selected type is 3 or Mexan\MxnTeam\Domain\Model\Team .

**Comment by Megafry** [comment]
Small Update:
tobenschmidt from the TYPO3 Slack channel ( [post](https://typo3.slack.com/archives/C0CRS7RFH/p1593426417333400) ) help me out.

 ```php
return [
  \Mexan\MxnTeam\Domain\Model\Team::class => [
      'tableName' => 'tx_news_domain_model_news',
      'recordType' => \Mexan\MxnTeam\Domain\Model\Team::class,
  ],
  \Mexan\MxnTeam\Domain\Model\Client::class => [
      'tableName' => 'tx_news_domain_model_news',
      'recordType' => \Mexan\MxnTeam\Domain\Model\Client::class,
  ],
  \GeorgRinger\News\Domain\Model\News::class => [
      'tableName' => 'tx_news_domain_model_news',
      //'recordType' => 0,
      'subclasses' => [
          \Mexan\MxnTeam\Domain\Model\Team::class,
          \Mexan\MxnTeam\Domain\Model\Client::class,
      ]
  ],
];
```

This works fine, even with 2 custom types.
but unfortunately the default news are no longer loaded
but if I add `recordType => 0`, then only normal news and my custom types are visible, but not the type 1 and 2 (Internal and external)

I updated the extension:
[mxn_team](https://github.com/Megafry/mxn_team/blob/master/Configuration/Extbase/Persistence/Classes.php#L17)

is there a way to prevent this?

![image (2)](https://user-images.githubusercontent.com/10438940/86018426-a13e6a80-ba25-11ea-9a93-9e540fec5df0.png)

**Comment by tobenschmidt** [comment]
I'm pretty sure that commit #1308 is the missing piece to make your solution work. As this commit is not included in the current release 8.3.0 you would need to patch news yourself or wait for the next release.

**Comment by georgringer** [comment]
closing it in favor of https://github.com/georgringer/news/pull/1308
